servlet/src/main/java/com/linecorp/armeria/server/servlet/ServletHttpExchange.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright 2020 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.server.servlet;
+
+import java.net.InetSocketAddress;
+import java.net.SocketAddress;
+
+import com.linecorp.armeria.common.AggregatedHttpRequest;
+import com.linecorp.armeria.common.HttpRequest;
+import com.linecorp.armeria.server.DefaultServiceRequestContext;
+import com.linecorp.armeria.server.ServiceRequestContext;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.Unpooled;
+import io.netty.handler.codec.http.DefaultFullHttpRequest;
+import io.netty.handler.codec.http.HttpMethod;
+import io.netty.handler.codec.http.HttpVersion;
+import io.netty.util.AttributeKey;
+
+/**
+ * Servlet object (contains 3 big objects: request, response, TCP channel).
+ */
+public final class ServletHttpExchange {
+    private static final AttributeKey<ServletHttpExchange> CHANNEL_ATTR_KEY_EXCHANGE =
+            AttributeKey.valueOf(ServletHttpExchange.class + "#ServletHttpExchange");
+
+    private ServletHttpRequest request;
+    private ServletHttpResponse response;
+    private DefaultServiceRequestContext channelHandlerContext;
+    private ServletContextImpl servletContext;
```

Comment:
How about passing and creating(if needed) all parameters so that we can make these all final?

Reply:
Can I refactor later?